benchmark/utils/DriverUtils.swift

```diff
@@ -407,6 +377,11 @@ func runBench(_ test: Test, _ c: TestConfig) -> BenchResults {
     } else {
       // Compute the scaling factor if a fixed c.fixedNumIters is not specified.
       scale = c.fixedNumIters
+      if scale == 1 { // FIXME: dirty fix. This method severly needs a refactoring
```

Comment:
It is not clear from the comment what the dirty fix is for...

Reply:
I was just venting… I'll remove the comment. The change correctly fixes the previously broken case when you specify `--num-iters=1`. I just very much dislike the code repetition. 

I hope to completely refactor it, right after demonstrating that my proposed approach works in `Benchmark_Driver`. If it works out, next step would be to move that logic down to `DriverUtils`, implying a total rework of this part of the code.